### ISSUE TITLE
cmake configure failure with netCDF

### ISSUE THREAD

**Issue by jandrej** [mfem/mfem#2243]
Something broke cmake with netCDF

```
-- NetCDF: looking for required package: HDF5, COMPONENTS: C;HL
CMake Warning (dev) at config/cmake/modules/MfemCmakeUtilities.cmake:450 (find_package):
  Policy CMP0074 is not set: find_package uses <PackageName>_ROOT variables.
  Run "cmake --help-policy CMP0074" for policy details.  Use the cmake_policy
  command to set the policy and suppress this warning.

  CMake variable HDF5_ROOT is set to:

    /Users/andrej1/local/hdf5-1.8.21

  For compatibility, CMake is ignoring the variable.
Call Stack (most recent call first):
  config/cmake/modules/FindNetCDF.cmake:27 (mfem_find_package)
  CMakeLists.txt:301 (find_package)
This warning is for project developers.  Use -Wno-dev to suppress it.

-- Found HDF5: hdf5::hdf5-shared (found version "1.8.21") found components: C HL
CMake Error at config/cmake/modules/MfemCmakeUtilities.cmake:525 (message):
   *** HDF5: unknown target. Please set HDF5_TARGET_NAMES.
Call Stack (most recent call first):
  config/cmake/modules/FindNetCDF.cmake:27 (mfem_find_package)
  CMakeLists.txt:301 (find_package)


-- Configuring incomplete, errors occurred!
See also "/Users/andrej1/repos/mfem/build-debug/CMakeFiles/CMakeOutput.log".
See also "/Users/andrej1/repos/mfem/build-debug/CMakeFiles/CMakeError.log".
```

**Comment by tzanio** [comment]
I believe @bslazarov may have run into this too.

Could it be due to https://github.com/mfem/mfem/pull/2141?

**Comment by bslazarov** [comment]
Can you check the log if the error is related to pthread library?

**Comment by jandrej** [comment]
@tzanio I built https://github.com/mfem/mfem/pull/2141 and netcdf is in my standard configure line so I think that worked.

@bslazarov It is not because of phtread.

It looks like the custom logic in mfem tries to find a "hdf5" target because of the package name but I haven't looked into it carefully.

ping @v-dobrev

**Comment by v-dobrev** [comment]
On my Mac with homebrew, HDF5 is found as a `<list of files>`:
```
-- Found HDF5: <list of files> (found version "1.12.0") found components: C HL 
```
In your case HDF5 is found as a target: `hdf5::hdf5-shared` -- I guess this comes from a cmake config file from the HDF5 installation you have. You can try to follow the suggestion:
```
   *** HDF5: unknown target. Please set HDF5_TARGET_NAMES.
```
Try setting `HDF5_TARGET_NAMES` to `hdf5` or `hdf5::hdf5-shared` or both.

We can try to make `mfem_find_package` smarter, so it does not require manual specification of the target name through `HDF5_TARGET_NAMES` but that will depend on what variables are defined by the cmake config file from your HDF5 installation.

**Comment by jandrej** [comment]
Setting `HDF5_TARGET_NAMES` results in
```
-- Imported HDF5[RELEASE]: /Users/andrej1/local/hdf5-1.8.21/lib/libhdf5.10.3.2.dylib
-- Found NetCDF: /Users/andrej1/local/netcdf-c-4.7.4/lib/libnetcdf.dylib;/Users/andrej1/local/hdf5-1.8.21/lib/libhdf5.10.3.2.dylib
-- NETCDF_INCLUDE_DIRS=/Users/andrej1/local/netcdf-c-4.7.4/include
CMake Error at config/cmake/modules/FindNetCDF.cmake:34 (list):
  list sub-command INSERT requires at least three arguments.
Call Stack (most recent call first):
```

**Comment by v-dobrev** [comment]
The line `config/cmake/modules/FindNetCDF.cmake:34` was introduced in #2141 -- it assumes the variable `HDF5_C_LIBRARY_hdf5_hl` is defined, which is true when using the `FindHDF5` module that comes with cmake, I suspect. I'm not sure if there is an equivalent variable when using the HDF5 defined cmake config files. If the imported target is properly defined, this `list(INSERT ...)` may actually be unnecessary even with static builds.

Can you try just commenting out that line in your case?

**Comment by jandrej** [comment]
Tried that, doesn't work.

**Comment by jandrej** [comment]
The culprit is a cmake build of HDF5 which doesn't correctly export targets, or cmakes own FindHDF5.cmake which is not communicating targets correctly.

**Comment by v-dobrev** [comment]
We can leave this issue open for now -- as a reminder to add support for HDF5's cmake config files, sometime in the future.

@bslazarov, are you also using hdf5 built with cmake? Or you see a different issue?

**Comment by bslazarov** [comment]
No, I am using HDF5 build with petsc. After the recent updates I cannot
compile mfem with netcdf/hdf5 libraries compiled with petsc. However, I
think  that my issue is different from this one.

**Comment by tzanio** [comment]
Is this issues still outstanding or can we close it now?

**Comment by v-dobrev** [comment]
> Is this issues still outstanding or can we close it now?

See https://github.com/mfem/mfem/issues/2243#issuecomment-841561952 above.

**Comment by bslazarov** [comment]
I will check it later today and I will let you know.

**Comment by bslazarov** [comment]
Yes it is still a problematic on my end. 

CMake Error at config/cmake/modules/FindNetCDF.cmake:34 (list):
  list sub-command INSERT requires at least three arguments.                                                      
Call Stack (most recent call first):                                                                              
  CMakeLists.txt:314 (find_package) 


It seems that  FindNetCDF utilize paths for HDF5 specified from somewhere outside than the MFEM configuration. I guess that if one provides an example in default.cmake  for these settings, a general user should be able to run the configuration.

**Comment by v-dobrev** [comment]
@bslazarov, I assume you are specifying explicitly the path to an HDF5 installation which then loads the HDF5 installed cmake files (like `hdf5-config.cmake`). These are the files that do not work correctly right now.

Can you try setting the option `-DHDF5_NO_FIND_PACKAGE_CONFIG_FILE=ON` (see https://cmake.org/cmake/help/latest/module/FindHDF5.html)?
